tools/codegen/core/gen_upb_api.sh

```diff
@@ -124,3 +124,18 @@ do
     --upb_out=$UPB_OUTPUT_DIR \
     --plugin=protoc-gen-upb=$UPB_PLUGIN
 done
+
+# In PHP build Makefile, the files with .upb.c suffix collide .upbdefs.c suffix due to a PHP buildsystem bug.
+# Work around this by placing the generated files with ".upbdefs.c" suffix under a different directory.
+# See https://github.com/grpc/grpc/issues/23307
+
+# move all .upbdefs.c files from under src/core/ext/upb-generated to src/core/ext/upbdefs-generated
+UPBDEFS_OUTPUT_DIR=$PWD/src/core/ext/upbdefs-generated
+rm -rf $UPBDEFS_OUTPUT_DIR
```

Comment:
Earlier in the script, there's an option of invoking this script and providing a custom location for the `upb-generated` directory name as `$1`. Just want to point out that there won't be such option for this new `upbdefs-generated` directory.

Reply:
Good point. I didn't want to get into details like that in the first iteration of the PR as it's just a technicality.
We could either:
- stop supporting the `$1` argument (which I'd be in favor of if this option it's used very often as these special cases add complexity to the script and I'd like to avoid unnecessary complexity as much as possible).
- make sure the $1 still works fine (doable, it's just a technicality).